app/src/main/java/ch/epfl/sweng/SDP/game/VotingPageActivity.java

```diff
@@ -36,15 +43,64 @@
     // For the moment it is defined as a constant
     private static final String PATH = "mockRooms.ABCDE";
     private static final String USER = "aa"; // need to be replaced with the username
+
+    private DatabaseReference rankingRef;
+    private DatabaseReference counterRef;
+    private DatabaseReference endTimeRef;
+    private DatabaseReference usersRef;
+    private DatabaseReference endVotingUsersRef;
+    private DatabaseReference stateRef;
+
+    private Bitmap[] drawings = new Bitmap[NUMBER_OF_DRAWINGS];
+    private short drawingDownloadCounter = 0;
+    private short changeDrawingCounter = 0;
+
+    private int[] ratings;
+    private short ratingToSendCounter = 0;
+
+    private String[] playersNames;
+
+    private ImageView drawingView;
+    private TextView playerNameView;
+    private TextView timer;
+    private RatingBar ratingBar;
+
+    private String roomID = "undefined";
+
     private static boolean enableAnimations = true;
-    private final ValueEventListener listenerEndUsersVoting = new ValueEventListener() {
+
+    public int[] getRatings() {
+        return ratings.clone();
+    }
+
+    protected final ValueEventListener listenerState = new ValueEventListener() {
         @Override
         public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-            Integer value = dataSnapshot.getValue(Integer.class);
+            Integer state = dataSnapshot.getValue(Integer.class);
+
+            if(state != null) {
+                GameStates stateEnum = GameStates.convertValueIntoState(state);
+                switch(stateEnum) {
+                    case END_VOTING_ACTIVITY:
+                        // Start ranking activity
```

Comment:
You have to use here `
// Prepare a Bundle for passing the ranking array to the fragment
                Bundle bundle = new Bundle();
                bundle.putStringArray("Ranking", ranking);

                // Clear the UI; buttonChangeImage and rankingButton need
                // to be removed after testing
                setVisibility(View.GONE, R.id.ratingBar, R.id.drawing, R.id.playerNameView,
                        R.id.buttonChangeImage, R.id.rankingButton);

                // Create and show the final ranking in the new fragment
                getSupportFragmentManager().beginTransaction()
                        .add(R.id.votingPageLayout,
                                RankingFragment.instantiate(getApplicationContext(),
                                        RankingFragment.class.getName(), bundle))
                        .addToBackStack(null).commit();`

Reply:
Will be implemented in a future pull request